kubemarine/plugins/calico.py

```diff
@@ -249,6 +306,25 @@ def enrich_deployment_calico_typha(self, manifest: Manifest) -> None:
             return None
         source_yaml = manifest.get_obj(key, patch=True)
 
+        if 'volumes' not in source_yaml['spec']['template']['spec']:
+            source_yaml['spec']['template']['spec']['volumes'] = []
+        if 'volumeMounts' not in source_yaml['spec']['template']['spec']['containers'][0]:
+            source_yaml['spec']['template']['spec']['containers'][0]['volumeMounts'] = []
+
+        source_yaml['spec']['template']['spec']['volumes'].append({
+        'name': 'calico-node-token',
+        'secret': {
+            'secretName': 'calico-node-token'  
+        }
+        })
+
+        source_yaml['spec']['template']['spec']['containers'][0]['volumeMounts'].append({
+        'name': 'calico-node-token',
+        'mountPath': '/var/run/secrets/kubernetes.io/serviceaccount'
+        })
+       
+        self.log.verbose(f"The {key} has been updated to include the new secret volume and mount.")
+
```

Comment:
There are quite many of boiler plate code. Consider implementing some generic method in manifest.Processor and re-use it.